testing/base/source_gen.cpp

```diff
@@ -386,35 +389,35 @@ static auto Sum(const T& range) -> int {
   return std::accumulate(range.begin(), range.end(), 0);
 }
 
-// A template function that implements the common logic of `GetIds` and
-// `GetUniqueIds`. Most parameters correspond to the parameters of those
-// functions. Additionally, an `AppendIds` callable is provided to implement the
-// appending operation.
+// A template function that implements the common logic of `GetIdentifiers` and
+// `GetUniqueIdentifiers`. Most parameters correspond to the parameters of those
+// functions. Additionally, an `AppendFunc` callable is provided to implement
+// the appending operation.
 //
 // The main functionality provided here is collecting the correct number of
 // identifiers from each of the lengths in the range [min_length, max_length]
 // and either in our default representative distribution or a uniform
 // distribution.
 //
 // Note that this template must be defined prior to its use below.
-template <typename AppendIds>
-auto SourceGen::GetIdsImpl(int number, int min_length, int max_length,
-                           bool uniform, AppendIds append_ids)
+template <typename AppendFunc>
```

Comment:
Very minor, but I think we're generally using `Fn` instead of `Func` as an abbreviation, consistent with Carbon.

But, and sorry if I'm missing context here, but had you considered something like `llvm::function_ref<int, int, llvm::SmallVector<llvm::StringRef>&>`?

Reply:
Good call on naming, but even better call on using `function_ref`, switched to that.

But using `SmallVectorImpl` to avoid encoding the small size (as above).